staging/src/k8s.io/apiserver/pkg/endpoints/handlers/response.go

```diff
@@ -49,18 +49,18 @@ func transformObject(ctx context.Context, obj runtime.Object, opts interface{},
 	case target == nil:
 		return obj, nil
 
-	case target.Kind == "PartialObjectMetadata" && target.GroupVersion() == metav1beta1.SchemeGroupVersion:
-		return asV1Beta1PartialObjectMetadata(obj)
+	case target.Kind == "PartialObjectMetadata":
+		return asPartialObjectMetadata(obj, target.GroupVersion())
 
-	case target.Kind == "PartialObjectMetadataList" && target.GroupVersion() == metav1beta1.SchemeGroupVersion:
-		return asV1Beta1PartialObjectMetadataList(obj)
+	case target.Kind == "PartialObjectMetadataList":
+		return asPartialObjectMetadataList(obj, target.GroupVersion())
 
-	case target.Kind == "Table" && target.GroupVersion() == metav1beta1.SchemeGroupVersion:
-		options, ok := opts.(*metav1beta1.TableOptions)
+	case target.Kind == "Table":
+		options, ok := opts.(*metav1.TableOptions)
```

Comment:
optionsForTransform below is still returning a `metav1beta1.TableOptions`... I'm not seeing how we get `metav1.TableOptions` objects here

Reply:
using metav1beta1 internally.  I wanted the new API up and running before I did the internal purge of metav1beta1 to be the default (since it goes into storage and tests and would have grown the PR non-trivially).

Reply:
that's fine... but how does the metav1beta1 options struct get switched to a metav1 options struct by the time we get here?

Reply:
I'm going to make this internally consistent (to reduce confusion) but type aliases are magical and you can cast either to both at runtime.

Reply:
Ah